### PR TITLE
System.Collections.* missed Equals nullable annotations 

### PR DESCRIPTION
**Pull request by hrrrrustic**

### CODE REVIEW

File: src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs [hunk 1]
```diff
@@ -143,7 +143,7 @@ public static partial class ImmutableArray
         public void CopyTo(T[] destination) { }
         public void CopyTo(T[] destination, int destinationIndex) { }
         public bool Equals(System.Collections.Immutable.ImmutableArray<T> other) { throw null; }
-        public override bool Equals(object? obj) { throw null; }
+        public override bool Equals([System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] object? obj) { throw null; }
```
Comment (eiriktsarpalis) [inline src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs:146]: This is not how the particular attribute is meant to be used. It is intended for aiding static analysis of `out` parameter nullability.

Comment (hrrrrustic) [reply src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs:146]: This is continuation of #47598, but if this kind of annotation is no longer relevant, you can close PR

Comment (eiriktsarpalis) [reply src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.cs:146]: Ah interesting, I had not realized the attribute could be used that way. I would say that it is by all means relevant.

**Review by eiriktsarpalis — APPROVED** [review]

**Comment by jeffhandley** [comment]
@eiriktsarpalis Is this ready to merge?

Once it's merged, please edit the description on https://github.com/dotnet/docs/issues/21202 to include the APIs affected by this PR, following the pattern of other PRs listed there (by preview).

**Pull request merged**